spanner/session.go

```diff
@@ -907,7 +907,7 @@ func (p *sessionPool) close(ctx context.Context) {
 
 func deleteSession(ctx context.Context, s *session, wg *sync.WaitGroup) {
 	defer wg.Done()
-	s.destroyWithContext(ctx, false)
+	s.destroyWithContext(ctx, false, true)
```

Comment:
Is `deleteSession` only called for sessions that are in use at that moment?

Note that `inUse` means that the session was checked out of the pool at the moment that this method is being called. So in this case it would mean that we are calling `deleteSession(..)` for a session that was checked out.

Reply:
But won't that then mean that the metric will drop to a negative value for a (very) short time when the application is shutting down? Assume that the situation is that:
1. The pool has 100 sessions.
2. 10 of them are in use.
3. The application shuts down and closes the client.
4. The client closes all sessions and calls this method for all 100 sesssions.
5. The inUseSessions metric drop from 10 to -90 for a very short time.

Reply:
wasInUse was set to false in latest commit, hence negative values should not appear.